DFC.ServiceTaxonomy.JobProfiles.Module/ServiceBusHandling/Converters/WhatItTakesMessageConverter.cs

```diff
@@ -25,39 +24,73 @@ public class WhatItTakesMessageConverter : IMessageConverter<WhatItTakesData>
         public async Task<WhatItTakesData> ConvertFromAsync(ContentItem contentItem)
         {
             var contentManager = _serviceProvider.GetRequiredService<IContentManager>();
-            //List<ContentItem> relatedSkills = Helper.GetContentItems(contentItem.Content.JobProfile.Relatedskills, contentManager);
+            IEnumerable<ContentItem> relatedSocSkillMatrixSkills = await Helper.GetContentItemsAsync(contentItem.Content.JobProfile.Relatedskills, contentManager);
             IEnumerable<ContentItem> relatedDigitalSkills = await Helper.GetContentItemsAsync(contentItem.Content.JobProfile.Digitalskills, contentManager);
             IEnumerable<ContentItem> relatedRestrictions = await Helper.GetContentItemsAsync(contentItem.Content.JobProfile.Relatedrestrictions, contentManager);
 
             var whatItTakesData = new WhatItTakesData
             {
-                //RelatedSkills = GetSkills(relatedSkills),
+                RelatedSocSkillMatrixSkills = GetSkills(relatedSocSkillMatrixSkills),
                 RelatedDigitalSkills = GetDigitalSkills(relatedDigitalSkills),
                 OtherRequirements = contentItem.Content.JobProfile.Otherrequirements.Html,
                 RelatedRestrictions = GetRestrictions(relatedRestrictions)
             };
             return whatItTakesData;
         }
 
-        // TODO: SKILLs
-        //private IEnumerable<SkillItem> GetSkills(List<ContentItem> contentItems)
-        //{
-        //    var skills = new List<SkillItem>();
-        //    if (contentItems.Any())
-        //    {
-        //        foreach (var contentItem in contentItems)
-        //        {
-        //            skills.Add(new SkillItem
-        //            {
-        //                Title = contentItem.As<TitlePart>().Title,
-        //                Description = contentItem.Content.Skill.Description?.Html,
-        //                ONetElementId = contentItem.Content.Skill.Description?.Text
-        //            });
-        //        }
-        //    }
+        private IEnumerable<SocSkillMatrixItem> GetSkills(IEnumerable<ContentItem> contentItems)
+        {
+            var skills = new List<SocSkillMatrixItem>();
+            if (contentItems.Any())
+            {
+                for (int i = 0; i < contentItems.Count(); i++)
+                {
+                    var skill = new SocSkillMatrixItem()
+                    {
+                        Id = contentItems.ElementAt(i).As<GraphSyncPart>().ExtractGuid(),
```

Comment:
Instead of using `contentItems.ElementAt(i)` multiple times, I'd rather have it assigned to a local variable and use that instead.

Reply:
Done. I have assigned contentItems.ElementAt(i) to local variable.